src/org/rascalmpl/library/Location.rsc

```diff
@@ -43,6 +43,63 @@ loc relativize(list[loc] haystack, loc needle) {
     }
 }
 
+@synopsis{Names of file system path syntaxes that can be found in the wild.}
+@description{
+* `generic` captures all unix-like systems like the BSD family, SunOs, Solaris, Irix, etc.
+* `mac_osx` captures modern Mac's that also have a unix-like filesystem but with different exceptions to permissable characters and case-sensitivity.
+* `windows` is for all DOS-based filesystems and what came after, with the slashes in the other direction.
+* `linux` is for all the filesystems in Linux distributions.
+
+Use this as a parameter to ((locFromFileSystem)) to choose how to parse a string as a filesystem path.
+}
+data FileSystemSyntax
+    = generic()
+    | mac_osx()
+    | windows()
+    | linux()
+    ;
+
+@javaClass{org.rascalmpl.library.Prelude}
+@synopsis{Utility to retrieve the current notation for file system paths.}
+@description{
+This is uses to configure the default syntax parameter of ((locFromFileSystem)).
+}
+java FileSystemSyntax currentFileSystem();
+
+@synopsis{Converts the OS-specific string representation of a file or directory PATH to a `loc`.}
+@description{
+This converts a string that hold a path to a file, written in a specific notation for paths, to
+a canonical `loc` in URI format.
+
+* if `legalize` is true, then every path segment is legalized by replacing offending characters to `_`
+* if `legalize` is false, and offending characters appear between the path separators, then an IO exception is thrown.
+* if the requested file system syntax is either case insensitive or not case preserving (or both), then all uppercase characters will be replaced by lowercase characters.
```

Comment:
We have to think about this syntax without an actual filesystem to test properties on. That's why this is hard.

I can turn it off, or make it configurable but only if the filesystem type supports it, or even allow the use to force normalized cases. In any case it will always be a problematic feature; without normalization can also be very bad the accuracy of downstream analysis on locs... 

Note that we do not normalize on windows because that would break the link with the actual filesystem.